pkg/diagnostics/http_tracing.go

```diff
@@ -32,12 +32,28 @@ const (
 
 var trimOWSRegExp = regexp.MustCompile(trimOWSRegexFmt)
 
-// SetTracingSpanContextFromHTTPContext sets the trace SpanContext in the request context
-func SetTracingSpanContextFromHTTPContext(next fasthttp.RequestHandler, spec config.TracingSpec) fasthttp.RequestHandler {
+// SetTracingInHTTPMiddleware sets the trace context or starts the trace client span based on request
+func SetTracingInHTTPMiddleware(next fasthttp.RequestHandler, appID string, spec config.TracingSpec) fasthttp.RequestHandler {
 	return func(ctx *fasthttp.RequestCtx) {
 		sc := GetSpanContextFromRequestContext(ctx, spec)
-		SpanContextToRequest(sc, &ctx.Request)
-		next(ctx)
+		path := string(ctx.Request.URI().Path())
+		method := ctx.Request.Header.Method()
+
+		if isAppChannelServiceInvocationRequest(appID, path) || isHealthzRequest(path) {
+			SpanContextToRequest(sc, &ctx.Request)
+			next(ctx)
+		} else {
+			spanName := fmt.Sprintf("%s:%s", method, path)
+
+			newCtx := NewContext((context.Context)(ctx), sc)
+			_, span := StartTracingClientSpanFromHTTPContext(newCtx, &ctx.Request, spanName, spec)
+			SpanContextToRequest(span.SpanContext(), &ctx.Request)
+			defer span.End()
```

Comment:
I think we do not need defer here. instead we can move line 51 to between L55 and L56.

Reply:
> I think we do not need defer here. instead we can move line 51 to between L55 and L56.

Agree , good catch !.